### ISSUE TITLE
KeyError from missing permalink

### ISSUE THREAD

**Issue by ghost** [ianpreston/redditfs#5]
```
(env)voltagex@icarus:~/src/redditfs$ python redditfs.py r
fusermount: failed to open /etc/fuse.conf: Permission denied
Traceback (most recent call last):
  File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 414, in _wrapper
    return func(*args, **kwargs) or 0
  File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 422, in getattr
    return self.fgetattr(path, buf, None)
  File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 668, in fgetattr
    attrs = self.operations('getattr', path.decode(self.encoding), fh)
  File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 709, in __call__
    return getattr(self, op)(*args)
  File "redditfs.py", line 41, in getattr
    f = self.traverse(path)
  File "redditfs.py", line 69, in traverse
    return self._traverse(path, node)
  File "redditfs.py", line 79, in _traverse
    self._lazy_load_subreddit(next_node, fn)
  File "redditfs.py", line 95, in _lazy_load_subreddit
    return self._populate_subreddit(filename)
  File "redditfs.py", line 127, in _populate_subreddit
    self._add_reddit_link_to_fs(root_file, zelda)
  File "redditfs.py", line 138, in _add_reddit_link_to_fs
    zelda['permalink']
KeyError: 'permalink'
```

**Comment by ghost** [comment]
Actually, it looks like the KeyError is something else entirely. Do you want me to open another issue for that?

**Comment by ianpreston** [comment]
Interesting, it looks like one of the links in a given subreddit doesn't have a permalink property in the response for some reason.

Is it reproducible for you? If so, could you add some debugging output for the subreddit that's being requested and the value of `zelda`? Adding `print fs.filename, zelda` between lines 132 and 133 might be sufficient.

**Comment by ghost** [comment]
```
    AACS {u'submit_text_html': None, u'user_is_banned': None, u'id': u'2wy69', u'submit_text': u'', u'display_name': u'AmericanAthletic', u'header_img': u'http://e.thumbs.redditmedia.com/i7XLLQUHxN27hcJF.png', u'description_html': u'&lt;!-- SC_OFF --&gt;&lt;div class="md"&gt;&lt;table&gt;&lt;thead&gt;\n&lt;tr&gt;\n&lt;th align="left"&gt;American BCS Rankings&lt;/th&gt;\n&lt;/tr&gt;\n&lt;/thead&gt;&lt;tbody&gt;\n&lt;/tbody&gt;&lt;/table&gt;\n\n&lt;p&gt;UCF #15, Louisville #18&lt;/p&gt;\n\n&lt;p&gt;Others Receiving Votes: Cincinnati&lt;/p&gt;\n\n&lt;hr/&gt;\n\n&lt;p&gt;American Athletic Football Standings:&lt;/p&gt;\n\n&lt;table&gt;&lt;thead&gt;\n&lt;tr&gt;\n&lt;th align="left"&gt;TEAM&lt;/th&gt;\n&lt;th align="center"&gt;CONF&lt;/th&gt;\n&lt;th align="center"&gt;OVERALL&lt;/th&gt;\n&lt;/tr&gt;\n&lt;/thead&gt;&lt;tbody&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;UCF&lt;/td&gt;\n&lt;td align="center"&gt;8-0&lt;/td&gt;\n&lt;td align="center"&gt;11-1&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Louisville&lt;/td&gt;\n&lt;td align="center"&gt;7-1&lt;/td&gt;\n&lt;td align="center"&gt;11-1&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Cincinnati&lt;/td&gt;\n&lt;td align="center"&gt;6-2&lt;/td&gt;\n&lt;td align="center"&gt;9-3&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Houston&lt;/td&gt;\n&lt;td align="center"&gt;5-3&lt;/td&gt;\n&lt;td align="center"&gt;8-4&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Southern Methodist&lt;/td&gt;\n&lt;td align="center"&gt;4-4&lt;/td&gt;\n&lt;td align="center"&gt;5-7&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Connecticut&lt;/td&gt;\n&lt;td align="center"&gt;3-5&lt;/td&gt;\n&lt;td align="center"&gt;3-9&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Rutgers&lt;/td&gt;\n&lt;td align="center"&gt;3-5&lt;/td&gt;\n&lt;td align="center"&gt;6-6&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;South Florida&lt;/td&gt;\n&lt;td align="center"&gt;2-6&lt;/td&gt;\n&lt;td align="center"&gt;2-10&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Memphis&lt;/td&gt;\n&lt;td align="center"&gt;1-7&lt;/td&gt;\n&lt;td align="center"&gt;3-9&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Temple&lt;/td&gt;\n&lt;td align="center"&gt;1-7&lt;/td&gt;\n&lt;td align="center"&gt;2-10&lt;/td&gt;\n&lt;/tr&gt;\n&lt;/tbody&gt;&lt;/table&gt;\n\n&lt;hr/&gt;\n\n&lt;table&gt;&lt;thead&gt;\n&lt;tr&gt;\n&lt;th align="left"&gt;American Basketball Coaches Poll Rankings&lt;/th&gt;\n&lt;/tr&gt;\n&lt;/thead&gt;&lt;tbody&gt;\n&lt;/tbody&gt;&lt;/table&gt;\n\n&lt;p&gt;Louisville #4, Connecticut #12, Memphis #15&lt;/p&gt;\n\n&lt;hr/&gt;\n\n&lt;p&gt;American Athletic Basketball Standings:&lt;/p&gt;\n\n&lt;table&gt;&lt;thead&gt;\n&lt;tr&gt;\n&lt;th align="left"&gt;TEAM&lt;/th&gt;\n&lt;th align="center"&gt;CONF&lt;/th&gt;\n&lt;th align="center"&gt;OVERALL&lt;/th&gt;\n&lt;/tr&gt;\n&lt;/thead&gt;&lt;tbody&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Cincinnati&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;7-1&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Connecticut&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;9-0&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Houston&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;6-4&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Louisville&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;8-1&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Rutgers&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;4-6&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Southern Methodist&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;8-2&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;South Florida&lt;/td&gt;\n&lt;td align="center"&gt;0-0&lt;/td&gt;\n&lt;td align="center"&gt;6-2&lt;/td&gt;\n&lt;/tr&gt;\n&lt;tr&gt;\n&lt;td align="left"&gt;Temple&lt;/td&gt;\n&lt;td align="centeTraceback (most recent call last):
      File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 414, in _wrapper
        return func(*args, **kwargs) or 0
      File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 422, in getattr
        return self.fgetattr(path, buf, None)
      File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 668, in fgetattr
        attrs = self.operations('getattr', path.decode(self.encoding), fh)
      File "/home/voltagex/src/redditfs/env/local/lib/python2.7/site-packages/fuse.py", line 709, in __call__
        return getattr(self, op)(*args)
      File "redditfs.py", line 41, in getattr
        f = self.traverse(path)
      File "redditfs.py", line 69, in traverse
        return self._traverse(path, node)
      File "redditfs.py", line 79, in _traverse
        self._lazy_load_subreddit(next_node, fn)
      File "redditfs.py", line 95, in _lazy_load_subreddit
        return self._populate_subreddit(filename)
      File "redditfs.py", line 127, in _populate_subreddit
        self._add_reddit_link_to_fs(root_file, zelda)
      File "redditfs.py", line 139, in _add_reddit_link_to_fs
        zelda['permalink']
    KeyError: 'permalink'
```

**Comment by ianpreston** [comment]
Looks like it wasn't handling redirects properly on the subset of missing subreddits that redirect to a search query. Should be fixed in 079f520e29def4616479307314b9e7a20721366f . Thanks!
